### ISSUE TITLE
emqx5.1.6 run error when install as a windows services

### ISSUE THREAD

**Issue by luchunminglu** [emqx/emqx#11537]
### What happened?

use bin\emqx install 安装为windows服务，启动服务，访问http://localhost:18083，无法访问，日志报错

### What did you expect to happen?

正常启动

### EMQX version

5.1.6

### OS version

windows server 2016

### Log files

2023-08-29T20:14:57.463000+08:00 [warning] msg: failed_to_resolve_env_variable, mfa: emqx_schema:naive_env_interpolation/1, line: 3259, env: {EMQX_LOG_DIR}, original: ${EMQX_LOG_DIR}/emqx.log
2023-08-29T20:14:57.500000+08:00 [warning] msg: failed_to_resolve_env_variable, mfa: emqx_schema:naive_env_interpolation/1, line: 3259, env: {EMQX_ETC_DIR}, original: ${EMQX_ETC_DIR}/acl.conf
2023-08-29T20:14:57.501000+08:00 [alert] msg: failed_to_read_acl_file, mfa: emqx_authz_file:validate/1, line: 53, explain: No such file or directory, path: ${EMQX_ETC_DIR}/acl.conf
2023-08-29T20:14:57.503000+08:00 [warning] msg: failed_to_resolve_env_variable, mfa: emqx_schema:naive_env_interpolation/1, line: 3259, env: {EMQX_LOG_DIR}, original: ${EMQX_LOG_DIR}/emqx.log
2023-08-29T20:14:57.511000+08:00 [error] crasher: initial call: application_master:init/4, pid: <0.638.0>, registered_name: [], exit: {{bad_return,{{emqx_conf_app,start,[normal,[]]},{'EXIT',{{case_clause,undefined},[{emqx_config_logger,tr_console_handler,1,[{file,"emqx_config_logger.erl"},{line,129}]},{emqx_config_logger,tr_handlers,1,[{file,"emqx_config_logger.erl"},{line,124}]},{emqx_config_logger,do_refresh_config,1,[{file,"emqx_config_logger.erl"},{line,48}]},{emqx_conf_app,start,2,[{file,"emqx_conf_app.erl"},{line,40}]},{application_master,start_it_old,4,[{file,"application_master.erl"},{line,293}]}]}}}},[{application_master,init,4,[{file,"application_master.erl"},{line,142}]},{proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,240}]}]}, ancestors: [<0.637.0>], message_queue_len: 1, messages: [{'EXIT',<0.639.0>,normal}], links: [<0.637.0>,<0.44.0>], dictionary: [], trap_exit: true, status: running, heap_size: 610, stack_size: 28, reductions: 190; neighbours:

**Comment by purusartha** [comment]
I have the exact problem. Any help will be appreciated

**Comment by id** [comment]
@luchunminglu @purusartha could you confirm if you used `bin/emqx install` or `bin/emqx.cmd install`?

**Comment by luchunminglu** [comment]
test with bin/emqx.cmd install and emqx.cmd install

**Comment by Fellerson** [comment]
Hello
I've the same issue, i've tried with .\bin\emqx.cmd install instead .\bin\emqx install but i've still the same issue on my side 

```
2023-09-25T10:07:08.659000+02:00 [alert] msg: failed_to_read_acl_file, mfa: emqx_authz_file:validate/1, line: 53, explain: No such file or directory, path: ${EMQX_ETC_DIR}/acl.conf
2023-09-25T10:07:08.662000+02:00 [warning] msg: failed_to_resolve_env_variable, mfa: emqx_schema:naive_env_interpolation/1, line: 3238, env: {EMQX_LOG_DIR}, original: ${EMQX_LOG_DIR}/emqx.log
2023-09-25T10:07:08.683000+02:00 [error] crasher: initial call: application_master:init/4, pid: <0.610.0>, registered_name: [], exit: {{bad_return,{{emqx_conf_app,start,[normal,[]]},{'EXIT',{{case_clause,undefined},[{emqx_config_logger,tr_console_handler,1,[{file,"emqx_config_logger.erl"},{line,129}]},{emqx_config_logger,tr_handlers,1,[{file,"emqx_config_logger.erl"},{line,124}]},{emqx_config_logger,do_refresh_config,1,[{file,"emqx_config_logger.erl"},{line,48}]},{emqx_conf_app,start,2,[{file,"emqx_conf_app.erl"},{line,40}]},{application_master,start_it_old,4,[{file,"application_master.erl"},{line,293}]}]}}}},[{application_master,init,4,[{file,"application_master.erl"},{line,142}]},{proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,240}]}]}, ancestors: [<0.609.0>], message_queue_len: 1, messages: [{'EXIT',<0.611.0>,normal}], links: [<0.609.0>,<0.44.0>], dictionary: [], trap_exit: true, status: running, heap_size: 610, stack_size: 28, reductions: 190; neighbours:
2023-09-25T10:08:53.376000+02:00 [warning] msg: failed_to_resolve_env_variable, mfa: emqx_schema:naive_env_interpolation/1, line: 3238, env: {EMQX_LOG_DIR}, original: ${EMQX_LOG_DIR}/emqx.log
2023-09-25T10:08:53.434000+02:00 [warning] msg: failed_to_resolve_env_variable, mfa: emqx_schema:naive_env_interpolation/1, line: 3238, env: {EMQX_ETC_DIR}, original: ${EMQX_ETC_DIR}/acl.conf
2023-09-25T10:08:53.434000+02:00 [alert] msg: failed_to_read_acl_file, mfa: emqx_authz_file:validate/1, line: 53, explain: No such file or directory, path: ${EMQX_ETC_DIR}/acl.conf
2023-09-25T10:08:53.436000+02:00 [warning] msg: failed_to_resolve_env_variable, mfa: emqx_schema:naive_env_interpolation/1, line: 3238, env: {EMQX_LOG_DIR}, original: ${EMQX_LOG_DIR}/emqx.log
2023-09-25T10:08:53.466000+02:00 [error] crasher: initial call: application_master:init/4, pid: <0.610.0>, registered_name: [], exit: {{bad_return,{{emqx_conf_app,start,[normal,[]]},{'EXIT',{{case_clause,undefined},[{emqx_config_logger,tr_console_handler,1,[{file,"emqx_config_logger.erl"},{line,129}]},{emqx_config_logger,tr_handlers,1,[{file,"emqx_config_logger.erl"},{line,124}]},{emqx_config_logger,do_refresh_config,1,[{file,"emqx_config_logger.erl"},{line,48}]},{emqx_conf_app,start,2,[{file,"emqx_conf_app.erl"},{line,40}]},{application_master,start_it_old,4,[{file,"application_master.erl"},{line,293}]}]}}}},[{application_master,init,4,[{file,"application_master.erl"},{line,142}]},{proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,240}]}]}, ancestors: [<0.609.0>], message_queue_len: 1, messages: [{'EXIT',<0.611.0>,normal}], links: [<0.609.0>,<0.44.0>], dictionary: [], trap_exit: true, status: running, heap_size: 610, stack_size: 28, reductions: 190; neighbours:
2023-09-25T10:11:05.414000+02:00 [warning] msg: failed_to_resolve_env_variable, mfa: emqx_schema:naive_env_interpolation/1, line: 3238, env: {EMQX_LOG_DIR}, original: ${EMQX_LOG_DIR}/emqx.log
2023-09-25T10:11:05.493000+02:00 [warning] msg: failed_to_resolve_env_variable, mfa: emqx_schema:naive_env_interpolation/1, line: 3238, env: {EMQX_ETC_DIR}, original: ${EMQX_ETC_DIR}/acl.conf
2023-09-25T10:11:05.494000+02:00 [alert] msg: failed_to_read_acl_file, mfa: emqx_authz_file:validate/1, line: 53, explain: No such file or directory, path: ${EMQX_ETC_DIR}/acl.conf
2023-09-25T10:11:05.497000+02:00 [warning] msg: failed_to_resolve_env_variable, mfa: emqx_schema:naive_env_interpolation/1, line: 3238, env: {EMQX_LOG_DIR}, original: ${EMQX_LOG_DIR}/emqx.log
2023-09-25T10:11:05.515000+02:00 [error] crasher: initial call: application_master:init/4, pid: <0.610.0>, registered_name: [], exit: {{bad_return,{{emqx_conf_app,start,[normal,[]]},{'EXIT',{{case_clause,undefined},[{emqx_config_logger,tr_console_handler,1,[{file,"emqx_config_logger.erl"},{line,129}]},{emqx_config_logger,tr_handlers,1,[{file,"emqx_config_logger.erl"},{line,124}]},{emqx_config_logger,do_refresh_config,1,[{file,"emqx_config_logger.erl"},{line,48}]},{emqx_conf_app,start,2,[{file,"emqx_conf_app.erl"},{line,40}]},{application_master,start_it_old,4,[{file,"application_master.erl"},{line,293}]}]}}}},[{application_master,init,4,[{file,"application_master.erl"},{line,142}]},{proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,240}]}]}, ancestors: [<0.609.0>], message_queue_len: 1, messages: [{'EXIT',<0.611.0>,normal}], links: [<0.609.0>,<0.44.0>], dictionary: [], trap_exit: true, status: running, heap_size: 610, stack_size: 28, reductions: 190; neighbours:
```

**Comment by ytchen2833** [comment]
Version 5.3.0 have the same issue.

```
2023-11-06T10:40:45.647000+08:00 [warning] msg: failed_to_resolve_env_variable, mfa: emqx_schema:naive_env_interpolation/1(3238), env: {EMQX_LOG_DIR}, original: ${EMQX_LOG_DIR}/emqx.log
2023-11-06T10:40:45.710000+08:00 [warning] msg: failed_to_resolve_env_variable, mfa: emqx_schema:naive_env_interpolation/1(3238), env: {EMQX_ETC_DIR}, original: ${EMQX_ETC_DIR}/acl.conf
2023-11-06T10:40:45.710000+08:00 [alert] msg: failed_to_read_acl_file, mfa: emqx_authz_file:validate/1(53), explain: No such file or directory, path: ${EMQX_ETC_DIR}/acl.conf
2023-11-06T10:40:45.712000+08:00 [warning] msg: failed_to_resolve_env_variable, mfa: emqx_schema:naive_env_interpolation/1(3238), env: {EMQX_LOG_DIR}, original: ${EMQX_LOG_DIR}/emqx.log
2023-11-06T10:40:45.722000+08:00 [error] crasher: initial call: application_master:init/4, pid: <0.609.0>, registered_name: [], exit: {{bad_return,{{emqx_conf_app,start,[normal,[]]},{'EXIT',{{case_clause,undefined},[{emqx_config_logger,tr_console_handler,1,[{file,"emqx_config_logger.erl"},{line,131}]},{emqx_config_logger,tr_handlers,1,[{file,"emqx_config_logger.erl"},{line,126}]},{emqx_config_logger,do_refresh_config,1,[{file,"emqx_config_logger.erl"},{line,50}]},{emqx_conf_app,start,2,[{file,"emqx_conf_app.erl"},{line,40}]},{application_master,start_it_old,4,[{file,"application_master.erl"},{line,293}]}]}}}},[{application_master,init,4,[{file,"application_master.erl"},{line,142}]},{proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,240}]}]}, ancestors: [<0.608.0>], message_queue_len: 1, messages: [{'EXIT',<0.610.0>,normal}], links: [<0.608.0>,<0.44.0>], dictionary: [], trap_exit: true, status: running, heap_size: 610, stack_size: 28, reductions: 190; neighbours:

```

**Comment by tiaan-lg** [comment]
I am experiencing similar issues when trying to setup a cluster running as a windows service

Any update on this issue?

**Comment by gemmtech** [comment]
I'm having the same issue and even tried previous 3 versions

Then reverted to trying virgin zip file, without making any changes and using previous older version 5.1.6, 5.2.1, 5.3.2 Still same issue as below.

2023-12-01T19:47:09.642000+00:00 [error] crasher: initial call: application_master:init/4, pid: <0.400.0>, registered_name: [], exit:

Any updates on fix?

**Comment by kedar-hatkar** [comment]
To solve issue of failed_to_resolve_env_variable EMQX_LOG_DIR and EMQX_ETC_DIR, 

add environment variables in windows
EMQX_LOG_DIR  = "Absolute path to emqx log directory"  eg "C:\emqx\log"
EMQX_ETC_DIR = "Absolute path to emqx etc directory"    eg "C:\emqx\log"

after adding environment variables and using default emqx.config file, i was able to run emqx as windows service and login in dashboard at http://localhost:18083.   but this technique is working on EMQX 5.0.26 to 5.3.1 version only. 

on version 5.3.2, i am also not able to run emqx as windows service. I have opened separate issue
https://github.com/emqx/emqx/issues/12090#issue-2023199904

**Comment by Fellerson** [comment]
> To solve issue of failed_to_resolve_env_variable EMQX_LOG_DIR and EMQX_ETC_DIR,
> 
> add environment variables in windows EMQX_LOG_DIR = "Absolute path to emqx log directory" eg "C:\emqx\log" EMQX_ETC_DIR = "Absolute path to emqx etc directory" eg "C:\emqx\etc"
> 
> after adding environment variables and using default emqx.config file, i was able to run emqx as windows service and login in dashboard at http://localhost:18083. but this technique is working on EMQX 5.0.26 to 5.3.1 version only.
> 
> on version 5.3.2, i am also not able to run emqx as windows service. I have opened separate issue [#12090 (comment)](https://github.com/emqx/emqx/issues/12090#issue-2023199904)

Hi

Thanks for your Reply. Tested on my side with EMQX 5.2.1 but it doesn't work (i've added both environment variables), unsucessfully.

**Comment by kedar-hatkar** [comment]
> > To solve issue of failed_to_resolve_env_variable EMQX_LOG_DIR and EMQX_ETC_DIR,
> > add environment variables in windows EMQX_LOG_DIR = "Absolute path to emqx log directory" eg "C:\emqx\log" EMQX_ETC_DIR = "Absolute path to emqx etc directory" eg "C:\emqx\etc"
> > after adding environment variables and using default emqx.config file, i was able to run emqx as windows service and login in dashboard at http://localhost:18083. but this technique is working on EMQX 5.0.26 to 5.3.1 version only.
> > on version 5.3.2, i am also not able to run emqx as windows service. I have opened separate issue [#12090 (comment)](https://github.com/emqx/emqx/issues/12090#issue-2023199904)
> 
> Hi
> 
> Thanks for your Reply. Tested on my side with EMQX 5.2.1 but it doesn't work (i've added both environment variables), unsucessfully.

Hi @Fellerson 

what error you are getting in emqx log file ? 

when i tried 5.2.1,i got error as [error] Generic server emqx_telemetry terminating. Reason: {'module could not be loaded'. 
But this error is fixed in version 5.3.0 as mentioned in release notes. Also i suggest to use latest version 5.3.2 ( if windows service approach not required ) or 5.3.1 ( windows service approach )

**Comment by Fellerson** [comment]
> > > To solve issue of failed_to_resolve_env_variable EMQX_LOG_DIR and EMQX_ETC_DIR,
> > > add environment variables in windows EMQX_LOG_DIR = "Absolute path to emqx log directory" eg "C:\emqx\log" EMQX_ETC_DIR = "Absolute path to emqx etc directory" eg "C:\emqx\etc"
> > > after adding environment variables and using default emqx.config file, i was able to run emqx as windows service and login in dashboard at http://localhost:18083. but this technique is working on EMQX 5.0.26 to 5.3.1 version only.
> > > on version 5.3.2, i am also not able to run emqx as windows service. I have opened separate issue [#12090 (comment)](https://github.com/emqx/emqx/issues/12090#issue-2023199904)
> > 
> > 
> > Hi
> > Thanks for your Reply. Tested on my side with EMQX 5.2.1 but it doesn't work (i've added both environment variables), unsucessfully.
> 
> Hi @Fellerson
> 
> what error you are getting in emqx log file ?
> 
> when i tried 5.2.1,i got error as [error] Generic server emqx_telemetry terminating. Reason: {'module could not be loaded'. But this error is fixed in version 5.3.0 as mentioned in release notes. Also i suggest to use latest version 5.3.2 ( if windows service approach not required ) or 5.3.1 ( windows service approach )

Hi

Thanks a lot for your answer

That's very strange. No logs are coming in the folder that i added in the environment paths.

**Comment by zmstone** [comment]
Windows support is dropped since 5.4.0.
Users are advised to run EMQX in docker on Windows.
